nilearn/decoding/space_net.py

```diff
@@ -835,11 +823,23 @@ def _set_coef_and_intercept(self, w):
         if self.w_.ndim == 1:
             self.w_ = self.w_[np.newaxis, :]
         self.coef_ = self.w_[:, :-1]
-        if self.is_classif:
+        if self._is_classification:
             self.intercept_ = self.w_[:, -1]
         else:
             self._set_intercept(self.Xmean_, self.ymean_, self.Xstd_)
```

Comment:
The same here. The base class would have `_set_intercept` method that is common to all inheriting classes. And classifier should implement its own behavior. If there is no such behavior, we can leave _set_intercept as abstract method and raise `NotImplemented` error and each class should implement its behavior.